include/ibamr/StokesFifthOrderWaveBcCoef.h

```diff
@@ -0,0 +1,279 @@
+                    SAMRAI::tbox::Pointer<SAMRAI::pdat::ArrayData<NDIM, double> >& bcoef_data,
+                    SAMRAI::tbox::Pointer<SAMRAI::pdat::ArrayData<NDIM, double> >& gcoef_data,
+                    const SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> >& variable,
+                    const SAMRAI::hier::Patch<NDIM>& patch,
+                    const SAMRAI::hier::BoundaryBox<NDIM>& bdry_box,
+                    double fill_time = 0.0) const override;
+
+    /*
+     * \brief Return how many cells past the edge or corner of the patch the
+     * object can fill.
+     *
+     * The "extension" used here is the number of cells that a boundary box
+     * extends past the patch in the direction parallel to the boundary.
+     *
+     * Note that the inability to fill the sufficient number of cells past the
+     * edge or corner of the patch may preclude the child class from being used
+     * in data refinement operations that require the extra data, such as linear
+     * refinement.
+     *
+     * The boundary box that setBcCoefs() is required to fill should not extend
+     * past the limits returned by this function.
+     */
+    SAMRAI::hier::IntVector<NDIM> numberOfExtensionsFillable() const override;
+
+    //\}
+
+private:
+    /*!
+     * \brief Default constructor.
+     *
+     * \note This constructor is not implemented and should not be used.
+     */
+    StokesFifthOrderWaveBcCoef() = delete;
+
+    /*!
+     * \brief Copy constructor.
+     *
+     * \note This constructor is not implemented and should not be used.
+     *
+     * \param from The value to copy to this object.
+     */
+    StokesFifthOrderWaveBcCoef(const StokesFifthOrderWaveBcCoef& from) = delete;
+
+    /*!
+     * \brief Assignment operator.
+     *
+     * \note This operator is not implemented and should not be used.
+     *
+     * \param that The value to assign to this object.
+     *
+     * \return A reference to this object.
+     */
+    StokesFifthOrderWaveBcCoef& operator=(const StokesFifthOrderWaveBcCoef& that) = delete;
+
+    /*!
+     * Get wave parameters from input db.
+     */
+    void getFromInput(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> db);
+
+    /*!
+     * Initialize Stokes coefficients.
+     */
+    void initStokesCoefficients();
+
+    /*!
+     * Get surface elevation at a specified horizontal position and time.
+     */
+    double getSurfaceElevation(double x, double time) const;
+
+    /*!
+     * Get velocity at a specified position and time.
+     */
+    double getVelocity(double x, double z_plus_d, double time) const;
+
+    /*!
+     * Book-keeping.
+     */
+    std::string d_object_name;
+
+    /*!
+     * Velocity component index.
+     */
+    int d_comp_idx = IBTK::invalid_index;
+
+    /*!
+     * muparser object for filling boundary conditions other than wave inlet.
+     */
+    IBTK::muParserRobinBcCoefs* d_muparser_bcs;
```

Comment:
There is no need to store a pointer here: you might need to use one if we call non-const methods on `d_muparser_bcs` but `setBcCoefs` is const, so its okay.